esmvaltool/diag_scripts/hydrology/hype.py

```diff
@@ -0,0 +1,117 @@
+"""PCR-GLOBWB diagnostic."""
+import logging
+from pathlib import Path
+
+import dask.array as da
+import iris
+
+import numpy
+# daskframe is possible if
+# pip install 'fsspec>=0.3.3'
+import pandas
+
+
+from esmvaltool.diag_scripts.shared import (ProvenanceLogger,
+                                            get_diagnostic_filename,
+                                            group_metadata, run_diagnostic)
+
+logger = logging.getLogger(Path(__file__).name)
+
+
+def get_provenance_record(attributes):
+    """Create a provenance record."""
+
+    ancestor_file = attributes['filename']
+
+    record = {
+        'caption': "Forcings for the Hype hydrological model.",
+        'domains': ['global'],
+        'authors': [
+            'pelupessy_inti',
+        ],
+        'projects': [
+            'ewatercycle',
+        ],
+        'references': [
+            'acknow_project',
+        ],
+        'ancestors': [ancestor_file],
+    }
+    return record
+
+
+def get_output_stem(attributes):
+    """ get output file stem, specific to HYPE """
+
+    short_to_stem = dict(tas="Tobs",
+                         tasmin="TMINobs",
+                         tasmax="TMAXobs",
+                         pr="Pobs")
+
+    shortname = attributes["short_name"]
+    if shortname in short_to_stem:
+        stem = short_to_stem[shortname]
+    else:
+        stem = Path(attributes['filename']).stem + '_hype'
+
+    return stem
+
+
+def get_data_times_and_ids(attributes):
+    """ get the data table to be written and the times and indices """
+    input_file = attributes['filename']
+
+    cube = iris.load_cube(input_file)
+
+    data = numpy.array(cube.core_data()).T
+
+    # ad-hoc fix of precipitation:
+    if attributes["short_name"] == "pr":
+        if attributes["units"] == "kg m-2 s-1":
```

Comment:
Is this needed? It looks like you already do this in the recipe with the convert units preprocessor?

Reply:
the unit conversion is not working yet..i have commented this out in the recipe file now, so this is needed